### PR TITLE
advancer: fix the incorrect gc safepoint behaviours

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/52082

Problem Summary:
1. Advancer didn't handle GC response correct, if set the safepoint smaller(Normally it won't happen, because the ts from advancer always closed to the current time(RPO <= 5min)) than current service safepoint, it still get the response without error. 
2. Set service safepoint to 0 won't delete the value.

### What changed and how does it work?
1. handle the response with smaller ts request.
2. add new interface to remove service safepoint.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix the issue that del task won't clean up service safepoint.
```

### CODE REVIEW

File: br/pkg/streamhelper/advancer_env.go [hunk 1]
```diff
@@ -46,7 +48,20 @@ type PDRegionScanner struct {
...
+	if err != nil {
+		return 0, errors.Annotate(err, "failed to block gc until")
+	}
+	if minimalSafePoint > at {
+		return 0, errors.Errorf("failed to block gc until, minimal safe point %d is greater than the target %d", minimalSafePoint, at)
+	}
+	return at, nil
+}
+
+func (c PDRegionScanner) RemoveGCSafepoint(ctx context.Context) error {
```
Comment (YuJuncen) [inline br/pkg/streamhelper/advancer_env.go:61]: Maybe rename it to `UnblockGC` to match the name of `BlockGCUntil`.

**Review by YuJuncen — APPROVED** [review]
rest lgtm

File: br/pkg/streamhelper/advancer_env.go (hunk 1 above)
Comment (3pointer) [reply br/pkg/streamhelper/advancer_env.go]: sure

**Review by Leavrth — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]

**Pull request merged**